include/mqtt.h

```diff
@@ -257,6 +257,12 @@ int __cheri_compartment("MQTT") mqtt_unsubscribe(Timeout    *t,
  * Fetch ACK and PUBLISH notifications on a given MQTT connection, and keep
  * the connection alive.
  *
+ * This function will invoke the callbacks passed to `mqtt_connect`.  The
+ * connection object is protected by a recursive mutex, so these callbacks can
+ * call additional publish and subscribe functions.  If doing so, care must be
+ * taken to ensure that the buffer is not exhausted.  Calling `mqtt_run` from a
```

Comment:
Not sure what "care must be taken to ensure that the buffer is not exhausted" means here? This buffer is neither passed by users nor exposed to users, so it would be nice to say how users can avoid exhausting it. Does this mean limiting the amount of recursive calls?